View Controllers/ListViewController.m

```diff
@@ -156,12 +214,49 @@ - (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForR
  - (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
      
      if (self.segmentIndex == 0) {
-         [self setView];
          return self.favorites.count;
      }
      else {
-         [self setView];
          return self.wishlist.count;
      }
- }
+     
+}
+
+- (void)didAddFavorite:(NSNotification*)notification {
+    GMSPlace* place = (GMSPlace*)notification.object;
+    
+    // add to the favorites
+    NSMutableArray<GMSPlace*>* favorites = [NSMutableArray arrayWithArray:self.favorites];
+    [favorites insertObject:place atIndex:0];
+    self.favorites = (NSArray*)favorites;
+    
+    // reload table
+    [self.tableView reloadData];
+    
+    // hide default label
+    if(self.segmentIndex == 0)
+    {
+        self.tableView.hidden = NO;
+        self.defaultView.hidden = YES;
+    }
+}
+
+- (void)didAddToWishlist:(NSNotification*)notification {
+    GMSPlace* place = (GMSPlace*)notification.object;
```

Comment:
nit - syntax here (and sorry if i've missed this before) - the star should go next to the variable name, not the type

e.g. GMSPlace *place 

Also when you're casting, note that we should have a space between the object and the *